### ISSUE TITLE
`struct-tag` for `validate` tag is too opinionated

### ISSUE THREAD

**Issue by OscarVanL** [mgechev/revive#1509]
**Describe the bug**
I use a validate package https://github.com/gookit/validate in my repo which makes use of the `validate` struct tag. Today the linter failed on `revive` claiming the struct tags were invalid. However, they were actually correct.

It's not entirely apparent from the [docs page](https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#struct-tag), but the validate rule is actually for the https://pkg.go.dev/github.com/go-playground/validator/v10 package.

In my opinion, this linter is too opinionated to have on by default - `validate` as a name is very generic and probably used by lots of packages.

To make matters worse, it appears there is no way to disable only the `validate` struct-tag rule but keep the others.

**To Reproduce**
1. Use `golangci-lint` with the `revive` linter enabled
2. Create a struct with a validator function that `gookit/validate` has that `go-playground/validator` doesn't. For example:
```go
type Foo struct {
	Version      int    `validate:"gt:0"`
    EnableFooBar bool
    FooType      string `filter:"trim" validate:"in:FOO,BAR,BAZ"`
	FooBarName   string   `filter:"trim" validate:"required_if:enableFooBar,true|max_len:255"`
	Name         string `filter:"trim" validate:"min_len:1|max_len:256"`
}
```

**Expected behavior**
I can think of a few possible options/changes that might make sense:
* Disable this rule by default and have it opt-in (since `go-playground/validator` is not part of the standard library)
* Only run these validation rules if `go-playground/validator` is found in the `go.mod` file
* Make the specific `struct-tag` rules configurable.
    * Currently I can only enable/disable `struct-tag` entirely, but I actually would like to use the rules for `json`, `yaml`, `datastore`, etc.
    * It would be great if these could be granularly enabled/disabled
* Document on the [struct-tag](https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#struct-tag) README which package each of the rules is for

**Logs**
The above example struct `Foo` will fail with errors like this:
```
foo.go: struct-tag: unknown option "gt:0" in validate tag (revive)
	Version      int       `validate:"gt:0"`

foo.go: struct-tag: unknown option "required_if:enableFooBar" in validate tag (revive)
	FooBarName  string   `filter:"trim" validate:"required_if:enableFooBar,true|max_len:255"`
	                              ^
foo.go: struct-tag: unknown option "min_len:1" in validate tag (revive)
	Name      string `filter:"trim" validate:"min_len:1|max_len:256"`

foo.go: struct-tag: unknown option "in:FOO" in validate tag (revive)
	FooType string `filter:"trim" validate:"in:FOO,BAR,BAZ"`
```

**Desktop (please complete the following information):**
 - OS: MacOS 15.6.1
 - Version of Go: 1.25.0

**Additional context**

```
golangci-lint --version                                                                                  
golangci-lint has version 2.4.0 built with go1.25.0
```

**Comment by alexandear** [comment]
Thanks for reporting this issue! I understand your frustration with the `struct-tag` rule being too opinionated about the `validate` tag. However, there are a couple of important points to clarify:

1. The `struct-tag` rule is already [disabled by default](https://github.com/mgechev/revive/blob/44e19927c08bcd69cdf4e4aebd4543ea0f3da237/config/config.go#L16) - it's not enabled unless you explicitly enable it in your configuration.
2. The rule is [already configurable](https://revive.run/r#struct-tag) - you can configure which specific struct tags to add.

We will improve the documentation. Thank you for the suggestion.

**Comment by OscarVanL** [comment]
Thanks for the quick reply!

> 1. The struct-tag rule is already [disabled by default](https://github.com/mgechev/revive/blob/44e19927c08bcd69cdf4e4aebd4543ea0f3da237/config/config.go#L16) - it's not enabled unless you explicitly enable it in your configuration.

You are right to point out that `struct-tag` is disabled by default, my mistake! I think what confused me is that I updated from an older version of `golangci-lint` (where the `validate` rule did not exist) to a newer version which automatically turned this rule on where it was not previously.

So to be specific, my issue isn't that `struct-tag` is enabled by default (as you point out, it's not), but rather that `struct-tag`'s `validate` rule is.

> 2. The rule is [already configurable](https://revive.run/r#struct-tag) - you can configure which specific struct tags to add.

I had a look at the config [here](https://revive.run/r#struct-tag) and tried changing it prior to opening this issue, but couldn't get it to exclude this rule.

The docs seemed to suggest that this config option was more for allowlisting certain struct tag values, rather than enabling/disabling specific rules:
> Example: To accept the inline option in JSON tags (and outline and gnu in BSON tags) you must provide the following configuration

> [rule.struct-tag]
> arguments = ["json,inline", "bson,outline,gnu"]

I tried setting something like this:
```
    revive:
      severity: warning
      rules:
        - name: struct-tag
          arguments:
            - asn1
            - bson
            - datastore
            - default
            - json
            - mapstructure
            - properties
            - protobuf
            - required
            - spanner
            - toml
            - url
            - xml
            - yaml
```

But I get this error:
```
ERRO [linters_context] setup revive: cannot configure rule: "struct-tag": invalid argument to the struct-tag rule. Expecting a string of the form key[,option]+, got json 
```

I can pass this validation check by following the config items with a comma, but then the job still fails on the `validate` tag even though it's not included on the list:
```
    revive:
      severity: warning
      rules:
        - name: struct-tag
          arguments:
            - asn1,
            - bson,
            - datastore,
            - default,
            - json,
            - mapstructure,
            - properties,
            - protobuf,
            - required,
            - spanner,
            - toml,
            - url,
            - xml,
            - yaml,
```

Maybe I am misunderstanding, please could you help me with what configuration I need to keep all the default rules but exclude the `validate` rule?

**Comment by chavacava** [comment]
> The docs seemed to suggest that this config option was more for allowlisting certain struct tag values, rather than enabling/disabling specific rules

Indeed, the configuration of the rule is for _allowlisting_ options of struct tags.

**Comment by hu3bi** [comment]
There should be a sane default to allow all possible tags from https://github.com/go-playground/validator.
Instead, you should be able to disallow certain tags.

If I want to enable revive for go-validator, I have to manually add 30+ tags and say this tag is allowed if the tag contains 

> `validate:"…
